### ISSUE TITLE
accuracy never go above 0.5

### ISSUE THREAD

**Issue by abdualhag** [aleksas/keras-fine-tune-inception#1]
Hi,
I worked out a similar version to your code which should do the same as yours but I run to a little issue, the accuracy never goes above 0.5. See below:

```
from keras.applications.inception_v3 import InceptionV3
from keras.preprocessing import image
from keras.models import Model
from keras.layers import Dense, GlobalAveragePooling2D
from keras.preprocessing.image import ImageDataGenerator
from keras import backend as K

train_path = 'cats-and-dogs/train'
valid_path = 'cats-and-dogs/valid'
test_path = 'cats-and-dogs/test'

img_width, img_height = 299, 299

if K.image_data_format() == 'channels_first':
    input_shape = (3, img_width, img_height)
else:
    input_shape = (img_width, img_height, 3)


train_batches = ImageDataGenerator().flow_from_directory(train_path, target_size=(img_width,img_height), classes=['dog', 'cat'], batch_size=10)
valid_batches = ImageDataGenerator().flow_from_directory(valid_path, target_size=(img_width,img_height), classes=['dog', 'cat'], batch_size=4)

base_model = InceptionV3(weights='imagenet', include_top=False)


# add a global spatial average pooling layer
x = base_model.output
x = GlobalAveragePooling2D()(x)
# let's add a fully-connected layer
x = Dense(1024, activation='relu')(x)
# and a logistic layer -- we have 2 classes
predictions = Dense(2, activation='softmax')(x)

# this is the model we will train
model = Model(inputs=base_model.input, outputs=predictions)


for layer in base_model.layers:
    layer.trainable = False


# compile the model (should be done *after* setting layers to non-trainable)
model.compile(optimizer='rmsprop', loss='categorical_crossentropy', metrics=['accuracy'])


# train the model on the new data for a few epochs
#model.fit_generator(...)

model.fit_generator(
    train_generator,
    steps_per_epoch=4,
    epochs=5,
    validation_data=validation_generator,
    validation_steps=4)

```
There was no point going further as the top layer accuracy did not go up. It would be most appreciate if you could tell me what might be the issue here. Thanks.

**Comment by aleksas** [comment]
It looks like you don't normalize input ranges. Your images are 0-255 values but inception is expecting -1:1  values. **(but 0:1, as is here, seams to work also) .**

Check these lines: https://github.com/aleksas/keras-fine-tune-inception/blob/36a466fab3e21ba2c0f2d3f09e14aaf829a0debb/fine_tune_inceptionv3.py#L94-L100

**[UPDATED]**

Inception normalization samples:

https://github.com/danielvarga/keras-finetuning/blob/de24104f330ea8192e5ff33eb2d15e9c95844eb3/dataset.py#L8-L12

and here

https://github.com/googlecodelabs/tensorflow-for-poets-2/blob/fbb773a30afa78700847f967fdfcbab2aa1d2360/scripts/retrain.py#L861-L873

https://github.com/googlecodelabs/tensorflow-for-poets-2/blob/fbb773a30afa78700847f967fdfcbab2aa1d2360/scripts/retrain.py#L962-L964

**Comment by abdualhag** [comment]
Sorry it took me this long to get back to you. I guess I did not say that the first time but your exact code does not raise the accuracy above 0.5 either and adding the lines from 94-100 to my code does not help also. What bothers me is that the preprocess function have different parameter than `train_datagen`  in your code. For example, there is no subtraction in yours. It would be good to know the exact equivalent to the preprocess function below in `ImageDataGenerator`:

```
def preprocess_input(x0):
    x = x0 / 255.
    x -= 0.5
    x *= 2.
return x
```
I am looking forward to hear back from you. Thanks

**Comment by aleksas** [comment]
That bothers me also. I couldn't find a way to add subtraction. Anyway it works for me(fortunately/unfortunately). I added environment setup instructions. That may also have a significant impact.

**Comment by abdualhag** [comment]
Sounds good. If I ever found what was the issue, I will let you know. Thanks
